be/src/storage/rowset/segment_v2/scalar_column_iterator.cpp

```diff
@@ -186,6 +188,53 @@ Status ScalarColumnIterator::next_batch(size_t* n, vectorized::Column* dst) {
     return Status::OK();
 }
 
+Status ScalarColumnIterator::next_batch(const vectorized::SparseRange& range, vectorized::Column* dst) {
+    size_t prev_bytes = dst->byte_size();
+    vectorized::SparseRangeIterator iter = range.new_iterator();
+    size_t end_ord = _page->first_ordinal() + _page->num_rows();
+    bool contain_deleted_row = (dst->delete_state() != DEL_NOT_SATISFIED);
+    vectorized::SparseRange read_range;
+
+    DCHECK_EQ(range.begin(), _current_ordinal);
+    while (iter.has_more()) {
+        if (_page->remaining() == 0 && iter.begin() == end_ord) {
```

Comment:
seems this branch can be removed?

Reply:
this branch means next row is in next page, so we can do `_load_next_page` directly to avoid `seek_page`, i will add some comments to make it easier to understand